### PR TITLE
fix keymap mode type error

### PR DESCRIPTION
**Pull request by kiyoon**
Fix type mismatch error introduced in #363 and reported in #365

### CODE REVIEW

File: lua/nvim-treesitter/textobjects/attach.lua [hunk 1]
```diff
@@ -12,8 +12,13 @@ local function make_dot_repeatable(fn)
...
 
 function M.make_attach(functions, submodule, keymap_modes, opts)
-  keymap_modes = keymap_modes or "n"
+  if type(keymap_modes) == "string" then
+    keymap_modes = { keymap_modes }
+  elseif type(keymap_modes) ~= "table" then
+    keymap_modes = { "n" }
+  end
   opts = opts or {}
+
```
Comment (scheatkode) [inline lua/nvim-treesitter/textobjects/attach.lua:21]: I tested with this fix but the issue persists. The error originates from `make_detach`.

**Review by scheatkode — COMMENTED** [review]
Hoping this helps.

File: lua/nvim-treesitter/textobjects/attach.lua (hunk 1 above)
Comment (kiyoon) [reply lua/nvim-treesitter/textobjects/attach.lua:21]: Did I make another mistake?

Comment (kiyoon) [reply lua/nvim-treesitter/textobjects/attach.lua:21]: I see, sorry maybe I need more sleep haha

Comment (scheatkode) [reply lua/nvim-treesitter/textobjects/attach.lua:21]: That's the right fix, just not the right location :sweat_smile:. `make_detach` is the origin of the error while you applied the fix for `make_attach`.

Comment (kiyoon) [reply lua/nvim-treesitter/textobjects/attach.lua:21]: I thought I changed the detach part but I was rushing. Sorry about that. Now with #367 it should be good right?

**Pull request merged**